.github/workflows/build-container-image.yml

```diff
@@ -0,0 +1,62 @@
+name: publish-container-image
+
+on:
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - created
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    env:
+      CONTAINER_REGISTRY_USERNAME: ${GITHUB_ACTOR}
+      CONTAINER_REGISTRY_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Cache Maven Packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('./**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: Build necessary modules
+        run: mvn install -pl context-monitoring -am
+
+      - name: Get short commit SHA
+        if: ${{ github.event_name == 'push' }}
+        id: get_short_commit_SHA
+        run: echo "::set-output name=sha8::$(echo ${GITHUB_SHA} | cut -c1-8)"
+
+      - name: Build Image
+        if: ${{ github.event_name == 'push' }}
+        env:
+          IMAGE_TAG: ${{ steps.get_short_commit_SHA.outputs.sha8}}
+        run: mvn compile jib:build -pl smartclide-monitoring -Djib.to.auth.username="${CONTAINER_REGISTRY_USERNAME}" -Djib.to.auth.password="${CONTAINER_REGISTRY_TOKEN}" -Djib.to.tags="${IMAGE_TAG}"
+
+      # below code triggers only for release events
+
+      - name: Get release name
+        if: ${{ github.event_name == 'release' }}
+        id: get_release_name
+        run: echo "::set-output name=RELEASE_NAME::${{ github.event.release.name }}"
```

Comment:
It would be better to use `github.event.release.tag_name` here. A release will always have a tag, but the name might be empty.

See https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#release and example release https://github.com/eclipse-researchlabs/smartclide-context/releases/edit/2.1.0